Refactor/search view models

hagarabobakr opened the pull request:
Modifying the SearchViewModel and FilterViewModel based on Barq's comments in the code review.

hagarabobakr committed:
rename navigation package to api

hagarabobakr committed:
rename navigation package to api

hagarabobakr committed:
refactor:
This commit reorganizes and reorders the search-related functions within the `SearchViewModel` for better clarity and maintainability.

The key changes include:
- Grouping related functions together.
- Moving private helper functions closer to their usage.
- Ensuring a more logical flow of operations within the ViewModel.

No functional changes were introduced in this refactoring.

hagarabobakr committed:
refactor SearchViewModel for better state management

This commit refactors the `SearchViewModel` to improve state management by:

- Introducing `setLoadingState` and `setSuccessState` helper functions to reduce boilerplate code for updating loading and success states.
- Modifying `onGetRecentViewedItems` and `getRecentSearchHistory` to return domain models directly, with UI mapping handled in `onCollectRecentViewedItems` and `onCollectRecentSearchHistory` respectively.
- Removing unused import `RecentSearchUiModel`.

hagarabobakr committed:
refactor: extract common paging logic in SearchViewModel

This commit refactors the `SearchViewModel` to extract common paging logic into a reusable `createPagingFlow` function. This simplifies the `loadMoviesOperation`, `loadTvShowsOperation`, and `loadActorsOperation` methods by removing redundant Pager configuration and mapping logic.

Additionally, the order of some methods within the class has been reorganized for better readability.

hagarabobakr committed:
refactor SearchViewModel to use helper functions for creating PagingSources

hagarabobakr committed:
Remove unnecessary code from SearchViewModel and move filter button logic to SearchScreenUiState

hagarabobakr committed:
add ActorMapper for search feature

hagarabobakr committed:
modify FilterViewModel to extend BaseViewModel with Unit

hagarabobakr committed:
refactor: make FilterBottomSheet composable self-contained

hagarabobakr committed:
Remove unused isFilterButtonVisible property from SearchViewModelTest

hagarabobakr committed:
add test for MovieMapper

hagarabobakr committed:
add RecentSearchMapper tests

hagarabobakr committed:
add tests for ActorMapper

hagarabobakr committed:
make BaseViewModel's updateState internal

hagarabobakr committed:
add unit tests for RecentViewedMapper

hagarabobakr committed:
add test for TvShowMapper

hagarabobakr committed:
feat: add navigation tests for search results

hagarabobakr committed:
Merge branch 'refs/heads/development' into refactor/search-view-model

# Conflicts:
#	feature/search/presentation/src/main/java/com/sanaa/presentation/screen/SearchViewModel.kt

Radwa85 commented:
fix ci issues

Radwa85 commented on feature/search/presentation/src/test/java/com/sanaa/presentation/screen/SearchViewModelTest.kt:
```diff
@@ -657,6 +655,53 @@ class SearchViewModelTest {
+        // Given
+        val viewed = RecentViewedUiModel(22, "url", MediaTypeUi.TV_SERIES, false)
+
+        // When
```
we don`t need these comments anymore

Radwa85 commented on feature/search/presentation/src/test/java/com/sanaa/presentation/screen/SearchViewModelTest.kt:
```diff
@@ -657,6 +655,53 @@ class SearchViewModelTest {
+        // Then
+        searchViewModel.effect.test {
+            val effect = awaitItem()
+            Truth.assertThat(effect).isEqualTo(SearchScreenEffects.NavigateToTvShowDetails(viewed.id))
```
you can import and use assertThat directly for consistency

hagarabobakr committed:
Update import paths for use cases in presentation layer

hagarabobakr committed:
Merge branch 'refs/heads/development' into refactor/search-view-model

# Conflicts:
#	feature/search/presentation/src/main/java/com/sanaa/presentation/filter_bottomsheet/FilterViewModel.kt

hagarabobakr committed:
add GenreMapperTest to verify mapping logic

hagarabobakr committed:
remove commented lines in ActorMapperTest

hagarabobakr committed:
test: add unit test for fetchGenres in FilterViewModel

- Verify that `fetchGenres()` correctly updates the UI state with mapped genres from both movie and TV series use cases.

hagarabobakr committed:
remove Arrange, Act, Assert comments in tests

hagarabobakr committed:
add unit tests for tv series details use case

This commit introduces comprehensive unit tests for the `ManageTvSeriesDetailsUseCase`. The tests cover the following functionalities:

- Verifying correct retrieval and mapping of TV series season details, including:
    - Season title
    - Number of episodes
    - Episode titles
    - Episode IMDb ratings
- Verifying correct retrieval and mapping of TV series reviews, including:
    - Review ID
    - Author name
    - User handle
    - Avatar URL
    - Rating
    - Content
    - Creation date

hagarabobakr committed:
Refactor: use `assertThat` directly in `SearchViewModelTest`

The `Truth.assertThat` is replaced with a direct import of `assertThat` for conciseness in the `SearchViewModelTest` file.
